radixdlt-java/src/main/java/com/radixdlt/client/application/RadixApplicationAPI.java

```diff
@@ -825,6 +768,62 @@ public UnsignedAtom buildAtomWithFee(List<ParticleGroup> particleGroups) {
 		return new UnsignedAtom(new Atom(allParticleGroups, metaData));
 	}
 
+	/**
+	 * Represents an atomic transaction to be committed to the ledger
+	 */
+	public final class Transaction {
+		private final ArrayList<Action> actions = new ArrayList<>();
+		private Transaction() {
+		}
+
+		/**
+		 * Execute an action within this transaction
+		 * @param action the action to execute
+		 */
+		public void execute(Action action) {
+			this.actions.add(action);
+		}
+
+		/**
+		 * Commit the transaction onto the ledger
+		 * @return the results of committing
+		 */
+		public Result commit() {
+			return buildDisconnectedResult(actions.toArray(new Action[0])).connect();
```

Comment:
Would be better to have `buildDisconnectedResult(Action...)` take an `Iterable`, as this particular use case takes an Iterable, converts it to an array, only to have buildDisconnectedResult turn it back into an Iterable again.

There is another use in `execute(Action)` below that would need to be wrapped in a singleton, but currently it's wrapped in an array reference anyway, so I don't think this is a strong negative.